File: app/models/playing_card.rb
class PlayingCard
  attr_reader :rank, :suit

  def initialize(rank, suit)
    @rank = rank
    @suit = suit
  end

  def as_json
    {
      rank: rank,
      suit: suit
    }
  end

  def self.from_json(data)
    new(data['rank'], data['suit'])
  end

  def ==(other)
    rank == other.rank && suit == other.suit
  end
end

File: app/javascript/models/PlayingCard.js
class PlayingCard {
  constructor(json) {
    this._rank = json.rank
    this._suit = json.suit
  }

  rank() {
    return this._rank
  }

  suit() {
    return this._suit
  }

  toString() {
    return `${this.suit().toLowerCase()}${this.rank().toLowerCase()}`
  }
}

export default PlayingCard
